src/Seaborn.jl

```diff
@@ -53,706 +11,77 @@ function __init__()
     copy!(seaborn, pyimport_conda("seaborn", "seaborn"))
 end
 
-macro delegate(f_list...)
-    blocks = Expr(:block)
-    for f in f_list
-        block = quote
-            function $(esc(f))(args...; kwargs...)
-                seaborn.$(f)(args...; kwargs...)
-            end
-        end
-        push!(blocks.args, block)
+#metaprogram to export and define all API functions
+# https://seaborn.pydata.org/api.html
+for function_name ∈ [
+    "relplot",
+    "scatterplot",
+    "lineplot",
+    "displot",
+    "histplot",
+    "kdeplot",
+    "ecdfplot",
+    "rugplot",
+    "catplot",
+    "stripplot",
+    "swarmplot",
+    "boxplot",
+    "violinplot",
+    "boxenplot",
+    "pointplot",
+    "barplot",
+    "countplot",
+    "lmplot",
+    "regplot",
+    "residplot",
+    "heatmap",
+    "clustermap",
+    "FacetGrid",
+    "pairplot",
+    "PairGrid",
+    "jointplot",
+    "JointGrid",
+    "set_theme",
+    "axes_style",
+    "set_style",
+    "plotting_context",
+    "set_context",
+    "set_color_codes",
+    "reset_defaults",
+    "reset_orig",
+    "set_palette",
+    "color_palette",
+    "husl_palette",
+    "hls_palette",
+    "cubehelix_palette",
+    "dark_palette",
+    "light_palette",
+    "diverging_palette",
+    "blend_palette",
+    "xkcd_palette",
+    "crayon_palette",
+    "mpl_palette",
+    "choose_colorbrewer_palette",
+    "choose_cubehelix_palette",
+    "choose_light_palette",
+    "choose_dark_palette",
+    "choose_diverging_palette",
+    "despine",
+    "move_legend",
+    "saturate",
+    "desaturate",
+    "set_hls_values",
+    "load_dataset",
+    "get_datset_names",
+    "get_data_home"
+]
+    #symbolize the function name
+    f = Symbol(function_name)
+    #export and define the function
+    @eval begin
+        export $f
+        $f(args...; kwargs...) = seaborn.$f(args...; kwargs...)
```

Comment:
This loses the docstrings.  See PyPlot for how you can export the Python docstring to Julia lazily: https://github.com/JuliaPy/PyPlot.jl/blob/67305af500a1fef26b9d127e099d110a102900f5/src/PyPlot.jl#L190

Reply:
(You'll have to copy PyPlot's `LazyHelp` type: https://github.com/JuliaPy/PyPlot.jl/blob/67305af500a1fef26b9d127e099d110a102900f5/src/PyPlot.jl#L33-L59)

Reply:
I would also call it as `seaborn.$function_name`, i.e. using the string, to suppress the attempt at auto-conversion.